### PR TITLE
Add/generic vector ingests

### PR DESCRIPTION
**Pull request by paridhi-parajuli**
**Summary:** Refactor code to support vector data ingest
Addresses [Issue](https://github.com/orgs/US-GHG-Center/projects/1/views/7?pane=issue&itemId=71962132)

## Changes

* Generalized code for vector data ingest and  removed `eis`-related functionalies.
* Change default value for `EXTERNAL_ROLE_ARN` passed to the vector ingest ECS
* Add configs for loading data into features database
* Add function to assume AWS IAM role and return temporary credentials

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: dags/veda_data_pipeline/groups/discover_group.py [hunk 1]
```diff
@@ -101,10 +103,17 @@ def subdag_discover(event={}):
...
+        task_id="parallel_run_process_generic_vectors",
+        trigger_dag_id="veda_generic_ingest_vector",
+        python_callable=get_files_to_process,
+    )
+
     # extra no-op, needed to run in dynamic mapping context
     end_discover = EmptyOperator(task_id="end_discover", trigger_rule=TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS,)
     
-    discover_from_s3 >> raster_vector_branching >> [run_process_raster, run_process_vector]
+    discover_from_s3 >> raster_vector_branching >> [run_process_raster, run_process_vector,run_process_generic_vector]
```
Comment (slesaad) [inline dags/veda_data_pipeline/groups/discover_group.py:115]:
```suggestion
    discover_from_s3 >> raster_vector_branching >> [run_process_raster, run_process_vector, run_process_generic_vector]
```

File: dags/veda_data_pipeline/veda_process_generic_vector_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    "filename_regex": "^(.*).tif$",
+    "discovery": "s3",
+    "datetime_range": "month",
+    "upload": false,
+    "cogify": false,
+    "discovered": 33,
+    "payload": "s3://veda-uah-sit-mwaa-853558080719/events/geoglam/s3_discover_output_6c46b57a-7474-41fe-977a-19d164531cdc.json"
+}	
+```
+- [Supports linking to external content](https://github.com/NASA-IMPACT/veda-data-pipelines)
```
Comment (slesaad) [inline dags/veda_data_pipeline/veda_process_generic_vector_pipeline.py:31]: can you update this comment to reflect vector?

File: dags/veda_data_pipeline/veda_process_generic_vector_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    "collection": "<collection_name>",
+    "prefix": "<prefix>/",
+    "bucket": "<bucket>",
+    "filename_regex": "<filename_regex>",
+    "discovery": "<s3>|cmr",
+    "datetime_range": "<month>|<day>",
+    "upload": "<false> | true",
+    "cogify": "false | true",
+    "payload": "<s3_uri_event_payload",
+}
```
Comment (slesaad) [inline dags/veda_data_pipeline/veda_process_generic_vector_pipeline.py:44]: is this still the same?

File: docker_tasks/generic_vector_ingest/handler.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    s3_objects = event_received["objects"]
+    status = list()
+
+    for s3_object in s3_objects:
+        href = s3_object["assets"]["default"]["href"] 
+        filename = href.split("/")[-1].split(".")[0]
+
+        # Use id template when collection is not provided in the conf
+        if collection_not_provided:
+            layer_name = payload_event["id_template"].format(filename)
```
Comment (slesaad) [inline docker_tasks/generic_vector_ingest/handler.py:182]:
```suggestion
            layer_name = payload_event.get("id_template", "{}").format(filename)
```

File: dags/veda_data_pipeline/veda_process_generic_vector_pipeline.py (hunk 3 above)
Comment (paridhi-parajuli) [reply dags/veda_data_pipeline/veda_process_generic_vector_pipeline.py]: modified it

**Review by slesaad — APPROVED** [review]
Looks good to me. @amarouane-ABDELHAK , @smohiudd , @ranchodeluxe , you guys ok with merging this?

File: README.md [hunk 5]
```diff
@@ -78,6 +78,66 @@ Currently, the client id and domain of an existing Cognito user pool programmati
...
+
+```
+## Configuration Fields Description
+- `collection`: The collection_id of the raster or vector data.
+- `bucket`: The name of the S3 bucket where the data is stored.
+- `prefix`: The location within the bucket where the files are to be discovered.
+- `filename_regex`: A regex expression used to filter files based on naming patterns.
+- `id_template`: The format used to create item identifiers in the system.
+- `vector`: Set to true to trigger the generic vector ingestion pipeline.
+- `vector_eis`: Set to true to trigger the vector ingestion pipeline.
```
Comment (smohiudd) [inline README.md:133]: Suggestion: Can this description say "Set to true to trigger the **EIS specific** vector ingestion pipeline" 

The way it's written now it's difficult to find the difference between this one and the generic vector

File: README.md [hunk 6]
```diff
@@ -78,6 +78,66 @@ Currently, the client id and domain of an existing Cognito user pool programmati
...
+
+
+## Pipeline Behaviour
+Since this pipeline can ingest both raster and vector data, the configuration can be modified accordingly. The `"vector": true` triggers the `generic_ingest_vector` dag. If the `collection` is provided, it uses the collection name as the table name for ingestion (recommended to use `append` extra_flag when the collection is provided). When no `collection` is provided, it uses the `id_template` and generates a table name by appending the actual ingested filename to the id_template (recommended to use `overwrite` extra flag).
+
+Setting `"vector_eis": true `will trigger the `ingest_vector` dag. If neither of these flags is set, the raster ingestion will be triggered, with the configuration typically looking like the raster ingestion example above.
```
Comment (smohiudd) [inline README.md:139]: Suggestion: Setting `"vector_eis": true `will trigger the EIS specific `ingest_vector` dag

**Review by smohiudd — APPROVED** [review]

**Comment by smohiudd** [comment]
I just added some general comments on documentation. The way the original EIS vector dag is labelled it's hard to know the difference between that and the generic. So in the docs would be good to be specific that the `ingest_vector` DAG has logic the is tailored to EIS Fire datasets only

**Review by amarouane-ABDELHAK — APPROVED** [review]

**Pull request merged**